### PR TITLE
fix(contacts): handle optional scim fields

### PR DESCRIPTION
**Pull request by sreenara**
# COMPLETES # NA

## This pull request addresses

Optional fields in the SCIM API response. Not all contacts will have all the fields populated and the SCIM API mandates only 3 mandatory fields. This causes a crash in our code.

## by making the following changes

Mark all fields except 3 mandatory ones as optional. Handle contact resolution logic accordingly.
Added new UTs to test this optional field logic in the resolution.

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

1. Create a custom contact and verify on Webex
2. Create a cloud contact and verify on Webex
3. Test the optional fields and minimal contact response from SCIM using new UTs

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/calling/src/Contacts/ContactsClient.ts [hunk 1]
```diff
@@ -272,16 +273,15 @@ export class ContactsClient implements IContacts {
...
-        const {sipAddresses} = filteredContact[SCIM_WEBEXIDENTITY_USER];
-        const firstName = filteredContact.name.givenName;
-        const lastName = filteredContact.name.familyName;
-        const manager = filteredContact[SCIM_ENTERPRISE_USER].manager.displayName;
-        const department = filteredContact[SCIM_ENTERPRISE_USER].department;
-
-        let avatarURL = '';
-        if (photos?.length) {
-          avatarURL = photos[0].value;
+        let sipAddresses;
```
Comment (adhmenon) [inline packages/calling/src/Contacts/ContactsClient.ts:276]: Are we using this variable anywhere else in the code? I don't seem to see it anywehere else apart from the next line.

File: packages/calling/src/Contacts/ContactsClient.ts [hunk 2]
```diff
@@ -272,16 +273,15 @@ export class ContactsClient implements IContacts {
...
+        if (filteredContact[SCIM_WEBEXIDENTITY_USER]) {
```
Comment (adhmenon) [inline packages/calling/src/Contacts/ContactsClient.ts:277]: Nit: Maybe we can use ternary operator here also.. but just a suggestion.

**Review by adhmenon — APPROVED** [review]
LGTM. Just a small minro comment.

File: packages/calling/src/common/types.ts [hunk 3]
```diff
@@ -231,24 +231,24 @@ interface Manager {
...
   userName: string;
-  active: boolean;
-  name: Name;
-  displayName: string;
-  emails: URIAddress[];
+  active?: boolean;
+  name?: Name;
+  displayName?: string;
+  emails?: URIAddress[];
   userType: string;
```
Comment (adhmenon) [inline packages/calling/src/common/types.ts:246]: Also, will this not be a build issue? I thought optionals are usually kept at the end of the method signature and not before a mandatory arguement..

File: packages/calling/src/Contacts/ContactsClient.ts (hunk 2 above)
Comment (sreenara) [reply packages/calling/src/Contacts/ContactsClient.ts:277]: @adhmenon the ternary operator doesn't work here. Something to do with the way typescript recognizes this object member. I initially tried with ternary but kept getting errors that SCIM_WEBEXIDENTITY_USER could be undefined.

**Pull request merged**